src/data/serializer/XYZ.js

```diff
@@ -62,13 +68,36 @@ Ext.define('GeoExt.data.serializer.XYZ', {
             var serialized = {
                 baseURL: source.getUrls()[0],
                 opacity: layer.getOpacity(),
+                imageExtension: this._getImageExtensionFromSource(source)
+                    || 'png',
                 resolutions: tileGrid.getResolutions(),
                 tileSize: ol.size.toSize(tileGrid.getTileSize()),
                 type: "OSM"
             };
             return serialized;
+        },
+
+        /**
+         * Returns the file extension from the url and compares it to whitelist.
+         * Sources with an tileUrlFunction are currently not supported.
+         *
+         * @private
+         * @param {ol.Source} source An ol.Source.XYZ.
+         * @return {String} The fileExtension or false if no one is found.
+         */
+        _getImageExtensionFromSource: function(source){
+            var url = source.getUrls()[0];
+            var lastThree = url.substr(url.length - 3);
+
+            if(Ext.isDefined(url) &&
```

Comment:
If you take my advice from above, this needs to be changed: `if (url && /* … */ )` should be enough